### PR TITLE
New function cluster_locs(): detect well clusters

### PR DESCRIPTION
**Pull request by florisvdh**

### CODE REVIEW

File: R/cluster.R [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+#' Locations that can not be clustered receive a unique cluster value.
+#'
+#' The function's code was partly inspired by unpublished code from Ivy Jansen.
+#'
+#' @param input A dataframe with X/Y coordinates, or an \code{sf} object of
+#' geometry type \code{POINT}.
+#' A typical input dataframe is the collected output of \code{\link{get_locs}}.
+#' @param max_dist The maximum geospatial distance between two points to make
+#' them belong to the same cluster.
+#' The default value is sensible when using \emph{meter} as unit of the
```
Comment (hansvancalster) [inline R/cluster.R]: Whether the default is sensible not only depends on the unit of measurement but also on the application at hand?

File: R/cluster.R [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+#' The X coordinate variable name; only considered when \code{input} is a
+#' dataframe.
+#' Defaults to \code{"x"}.
+#' @param yvar String.
+#' The Y coordinate variable name; only considered when \code{input} is a
+#' dataframe.
+#' Defaults to \code{"y"}.
+#'
+#' @return
+#' The original object with an extra variable added to define
```
Comment (hansvancalster) [inline R/cluster.R]: Best add the name of the variable. It is named `output_var` in the code, but maybe a better name is `cluster_id`

File: R/cluster.R (hunk 1 above)
Comment (florisvdh) [reply R/cluster.R]: True, will add that. It has been tried (and tested with @w-jan) to cover most common usecases.

File: R/cluster.R (hunk 2 above)
Comment (florisvdh) [reply R/cluster.R]: Well, actually `output_var` is the argument name; by default the name `cluster` was set. Will clarify that default. But still, `cluster_id` is a better default, will replace!

**Pull request merged**